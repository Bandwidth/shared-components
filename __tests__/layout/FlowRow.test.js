import React from 'react';
import toJSON from 'enzyme-to-json';
import {
  FlowRow,
} from '../../src';

describe('the FlowRow component', () => {
  describe('rendering two items with labels', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowWithTheme((
        <FlowRow>
          <FlowRow.Item label="foo" />
          <FlowRow.Item label="bar" />
        </FlowRow>
      ));
    });

    test('renders items with labels', () => {
      const container = wrapper.find(FlowRow.Container);
      const nodes = container.find(FlowRow.Item);
      expect(container.props().suppressLabels).toBe(false);
      expect(nodes).toHaveLength(2);
    });
  });

  describe('rendering two items without labels', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowWithTheme((
        <FlowRow>
          <FlowRow.Item />
          <FlowRow.Item />
        </FlowRow>
      ));
    });

    test('renders items with suppressed labels', () => {
      const container = wrapper.find(FlowRow.Container);
      const nodes = container.find(FlowRow.Item);
      expect(container.props().suppressLabels).toBe(true);
      expect(nodes).toHaveLength(2);
    });
  });

  describe('rendering one item with a label and the other without', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowWithTheme((
        <FlowRow>
          <FlowRow.Item label="foo" />
          <FlowRow.Item />
        </FlowRow>
      ));
    });

    test('renders items with labels', () => {
      const container = wrapper.find(FlowRow.Container);
      const nodes = container.find(FlowRow.Item);
      expect(container.props().suppressLabels).toBe(false);
      expect(nodes).toHaveLength(2);
    });
  });
});

describe('the FlowRow component styling', () => {
  test('with two labeled items', () => {
    const wrapper = mountWithTheme((
      <FlowRow>
        <FlowRow.Item label="foo" />
        <FlowRow.Item label="bar" />
      </FlowRow>
    ));
    expect(toJSON(wrapper)).toMatchStyledComponentsSnapshot();
  });
  test('with two unlabeled items', () => {
    const wrapper = mountWithTheme((
      <FlowRow>
        <FlowRow.Item />
        <FlowRow.Item />
      </FlowRow>
    ));
    expect(toJSON(wrapper)).toMatchStyledComponentsSnapshot();
  });
  test('with one labeled item and one not', () => {
    const wrapper = mountWithTheme((
      <FlowRow>
        <FlowRow.Item label="foo" />
        <FlowRow.Item />
      </FlowRow>
    ));
    expect(toJSON(wrapper)).toMatchStyledComponentsSnapshot();
  });
});
