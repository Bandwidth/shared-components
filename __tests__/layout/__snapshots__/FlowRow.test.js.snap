// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the FlowRow component styling with one labeled item and one not 1`] = `
.gVvYyj > .FlowItem__Container-krUYur,
.gVvYyj > .FlowRow__BaseStyles-jBvvfh {
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.gVvYyj > .FlowItem__Container-krUYur:first-child,
.gVvYyj > .FlowRow__BaseStyles-jBvvfh:first-child {
  -webkit-flex-basis: calc(50% - 15px);
  -ms-flex-basis: calc(50% - 15px);
  flex-basis: calc(50% - 15px);
  margin-right: 15px;
}

.gVvYyj > .FlowItem__Container-krUYur:last-child,
.gVvYyj > .FlowRow__BaseStyles-jBvvfh:last-child {
  -webkit-flex-basis: calc(50% - 15px);
  -ms-flex-basis: calc(50% - 15px);
  flex-basis: calc(50% - 15px);
  margin-left: 15px;
}

.gVvYyj > .FlowItem__Container-krUYur:only-child,
.gVvYyj > .FlowRow__BaseStyles-jBvvfh:only-child {
  -webkit-flex-basis: 100%;
  -ms-flex-basis: 100%;
  flex-basis: 100%;
  margin-left: 0;
  margin-right: 0;
}

.gbwVLv {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.jVAKAe {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.jVAKAe > .Label__Label-jsURsk {
  height: 21px;
  margin: 0;
  margin-bottom: 5.6px;
  overflow-x: hidden;
  -webkit-flex-basis: 21px;
  -ms-flex-basis: 21px;
  flex-basis: 21px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__Content-bZFRMT {
  height: 53px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.jVAKAe > .FlowItem__Content-bZFRMT > * {
  margin: auto;
}

.jVAKAe > .FlowItem__Content-bZFRMT > *:not(:last-child):not(:only-child) {
  margin-right: 30px;
}

.jVAKAe > .FlowItem__FlexibleContent-kAxlQq {
  height: auto;
  -webkit-flex-basis: 53px;
  -ms-flex-basis: 53px;
  flex-basis: 53px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.jVAKAe > .HelpText__HelpText-cZQOvv {
  min-height: 18px;
  margin: 0;
  margin-top: 5.6px;
  -webkit-flex-basis: 18px;
  -ms-flex-basis: 18px;
  flex-basis: 18px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__MoreContent-eNWnOb {
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.gdcnMF {
  font-size: 1em;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  font-weight: 600;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
  padding-bottom: 0.4em;
  color: #666;
  background: transparent;
  opacity: 1;
  display: block;
  line-height: 1.5em;
}

<FlowRow
  __styled-components__={[Function]}
>
  <FlowRow__Styles
    suppressLabels={false}
  >
    <FlowRow__BaseStyles
      className="FlowRow__Styles-iRxcTN gVvYyj"
      suppressLabels={false}
    >
      <div
        className="FlowRow__Styles-iRxcTN gVvYyj FlowRow__BaseStyles-jBvvfh gbwVLv"
      >
        <FlowItem
          alignment="stretch"
          className={null}
          error={false}
          flexibleContent={false}
          helpText={null}
          id={null}
          label="foo"
          moreContent={null}
        >
          <FlowItem__Container
            alignment="stretch"
            className={null}
            id={null}
          >
            <div
              className="FlowItem__Container-krUYur jVAKAe"
              id={null}
            >
              <Label__Label
                className={null}
                disabled={false}
                id={null}
              >
                <label
                  className="Label__Label-jsURsk gdcnMF"
                  disabled={false}
                  id={null}
                >
                  foo
                </label>
              </Label__Label>
              <FlowItem__Content>
                <div
                  className="FlowItem__Content-bZFRMT cwqVhq"
                />
              </FlowItem__Content>
            </div>
          </FlowItem__Container>
        </FlowItem>
        <FlowItem
          alignment="stretch"
          className={null}
          error={false}
          flexibleContent={false}
          helpText={null}
          id={null}
          label={null}
          moreContent={null}
        >
          <FlowItem__Container
            alignment="stretch"
            className={null}
            id={null}
          >
            <div
              className="FlowItem__Container-krUYur jVAKAe"
              id={null}
            >
              <Label__Label
                className={null}
                disabled={false}
                id={null}
              >
                <label
                  className="Label__Label-jsURsk gdcnMF"
                  disabled={false}
                  id={null}
                />
              </Label__Label>
              <FlowItem__Content>
                <div
                  className="FlowItem__Content-bZFRMT cwqVhq"
                />
              </FlowItem__Content>
            </div>
          </FlowItem__Container>
        </FlowItem>
      </div>
    </FlowRow__BaseStyles>
  </FlowRow__Styles>
</FlowRow>
`;

exports[`the FlowRow component styling with two labeled items 1`] = `
.gVvYyj > .FlowItem__Container-krUYur,
.gVvYyj > .FlowRow__BaseStyles-jBvvfh {
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.gVvYyj > .FlowItem__Container-krUYur:first-child,
.gVvYyj > .FlowRow__BaseStyles-jBvvfh:first-child {
  -webkit-flex-basis: calc(50% - 15px);
  -ms-flex-basis: calc(50% - 15px);
  flex-basis: calc(50% - 15px);
  margin-right: 15px;
}

.gVvYyj > .FlowItem__Container-krUYur:last-child,
.gVvYyj > .FlowRow__BaseStyles-jBvvfh:last-child {
  -webkit-flex-basis: calc(50% - 15px);
  -ms-flex-basis: calc(50% - 15px);
  flex-basis: calc(50% - 15px);
  margin-left: 15px;
}

.gVvYyj > .FlowItem__Container-krUYur:only-child,
.gVvYyj > .FlowRow__BaseStyles-jBvvfh:only-child {
  -webkit-flex-basis: 100%;
  -ms-flex-basis: 100%;
  flex-basis: 100%;
  margin-left: 0;
  margin-right: 0;
}

.gbwVLv {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.jVAKAe {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.jVAKAe > .Label__Label-jsURsk {
  height: 21px;
  margin: 0;
  margin-bottom: 5.6px;
  overflow-x: hidden;
  -webkit-flex-basis: 21px;
  -ms-flex-basis: 21px;
  flex-basis: 21px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__Content-bZFRMT {
  height: 53px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.jVAKAe > .FlowItem__Content-bZFRMT > * {
  margin: auto;
}

.jVAKAe > .FlowItem__Content-bZFRMT > *:not(:last-child):not(:only-child) {
  margin-right: 30px;
}

.jVAKAe > .FlowItem__FlexibleContent-kAxlQq {
  height: auto;
  -webkit-flex-basis: 53px;
  -ms-flex-basis: 53px;
  flex-basis: 53px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.jVAKAe > .HelpText__HelpText-cZQOvv {
  min-height: 18px;
  margin: 0;
  margin-top: 5.6px;
  -webkit-flex-basis: 18px;
  -ms-flex-basis: 18px;
  flex-basis: 18px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__MoreContent-eNWnOb {
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.gdcnMF {
  font-size: 1em;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  font-weight: 600;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
  padding-bottom: 0.4em;
  color: #666;
  background: transparent;
  opacity: 1;
  display: block;
  line-height: 1.5em;
}

<FlowRow
  __styled-components__={[Function]}
>
  <FlowRow__Styles
    suppressLabels={false}
  >
    <FlowRow__BaseStyles
      className="FlowRow__Styles-iRxcTN gVvYyj"
      suppressLabels={false}
    >
      <div
        className="FlowRow__Styles-iRxcTN gVvYyj FlowRow__BaseStyles-jBvvfh gbwVLv"
      >
        <FlowItem
          alignment="stretch"
          className={null}
          error={false}
          flexibleContent={false}
          helpText={null}
          id={null}
          label="foo"
          moreContent={null}
        >
          <FlowItem__Container
            alignment="stretch"
            className={null}
            id={null}
          >
            <div
              className="FlowItem__Container-krUYur jVAKAe"
              id={null}
            >
              <Label__Label
                className={null}
                disabled={false}
                id={null}
              >
                <label
                  className="Label__Label-jsURsk gdcnMF"
                  disabled={false}
                  id={null}
                >
                  foo
                </label>
              </Label__Label>
              <FlowItem__Content>
                <div
                  className="FlowItem__Content-bZFRMT cwqVhq"
                />
              </FlowItem__Content>
            </div>
          </FlowItem__Container>
        </FlowItem>
        <FlowItem
          alignment="stretch"
          className={null}
          error={false}
          flexibleContent={false}
          helpText={null}
          id={null}
          label="bar"
          moreContent={null}
        >
          <FlowItem__Container
            alignment="stretch"
            className={null}
            id={null}
          >
            <div
              className="FlowItem__Container-krUYur jVAKAe"
              id={null}
            >
              <Label__Label
                className={null}
                disabled={false}
                id={null}
              >
                <label
                  className="Label__Label-jsURsk gdcnMF"
                  disabled={false}
                  id={null}
                >
                  bar
                </label>
              </Label__Label>
              <FlowItem__Content>
                <div
                  className="FlowItem__Content-bZFRMT cwqVhq"
                />
              </FlowItem__Content>
            </div>
          </FlowItem__Container>
        </FlowItem>
      </div>
    </FlowRow__BaseStyles>
  </FlowRow__Styles>
</FlowRow>
`;

exports[`the FlowRow component styling with two unlabeled items 1`] = `
.gYaPrd > .FlowItem__Container-krUYur,
.gYaPrd > .FlowRow__BaseStyles-jBvvfh {
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.gYaPrd > .FlowItem__Container-krUYur:first-child,
.gYaPrd > .FlowRow__BaseStyles-jBvvfh:first-child {
  -webkit-flex-basis: calc(50% - 15px);
  -ms-flex-basis: calc(50% - 15px);
  flex-basis: calc(50% - 15px);
  margin-right: 15px;
}

.gYaPrd > .FlowItem__Container-krUYur:last-child,
.gYaPrd > .FlowRow__BaseStyles-jBvvfh:last-child {
  -webkit-flex-basis: calc(50% - 15px);
  -ms-flex-basis: calc(50% - 15px);
  flex-basis: calc(50% - 15px);
  margin-left: 15px;
}

.gYaPrd > .FlowItem__Container-krUYur:only-child,
.gYaPrd > .FlowRow__BaseStyles-jBvvfh:only-child {
  -webkit-flex-basis: 100%;
  -ms-flex-basis: 100%;
  flex-basis: 100%;
  margin-left: 0;
  margin-right: 0;
}

.gYaPrd .Label__Label-jsURsk {
  display: none;
}

.gbwVLv {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.jVAKAe {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.jVAKAe > .Label__Label-jsURsk {
  height: 21px;
  margin: 0;
  margin-bottom: 5.6px;
  overflow-x: hidden;
  -webkit-flex-basis: 21px;
  -ms-flex-basis: 21px;
  flex-basis: 21px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__Content-bZFRMT {
  height: 53px;
  -webkit-flex-grow: 0;
  -ms-flex-grow: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.jVAKAe > .FlowItem__Content-bZFRMT > * {
  margin: auto;
}

.jVAKAe > .FlowItem__Content-bZFRMT > *:not(:last-child):not(:only-child) {
  margin-right: 30px;
}

.jVAKAe > .FlowItem__FlexibleContent-kAxlQq {
  height: auto;
  -webkit-flex-basis: 53px;
  -ms-flex-basis: 53px;
  flex-basis: 53px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}

.jVAKAe > .HelpText__HelpText-cZQOvv {
  min-height: 18px;
  margin: 0;
  margin-top: 5.6px;
  -webkit-flex-basis: 18px;
  -ms-flex-basis: 18px;
  flex-basis: 18px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
  padding: 0;
}

.jVAKAe > .FlowItem__MoreContent-eNWnOb {
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.gdcnMF {
  font-size: 1em;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  font-weight: 600;
  font-family: "Overpass", Raleway, "Open Sans", arial, sans-serif;
  padding-bottom: 0.4em;
  color: #666;
  background: transparent;
  opacity: 1;
  display: block;
  line-height: 1.5em;
}

<FlowRow
  __styled-components__={[Function]}
>
  <FlowRow__Styles
    suppressLabels={true}
  >
    <FlowRow__BaseStyles
      className="FlowRow__Styles-iRxcTN gYaPrd"
      suppressLabels={true}
    >
      <div
        className="FlowRow__Styles-iRxcTN gYaPrd FlowRow__BaseStyles-jBvvfh gbwVLv"
      >
        <FlowItem
          alignment="stretch"
          className={null}
          error={false}
          flexibleContent={false}
          helpText={null}
          id={null}
          label={null}
          moreContent={null}
        >
          <FlowItem__Container
            alignment="stretch"
            className={null}
            id={null}
          >
            <div
              className="FlowItem__Container-krUYur jVAKAe"
              id={null}
            >
              <Label__Label
                className={null}
                disabled={false}
                id={null}
              >
                <label
                  className="Label__Label-jsURsk gdcnMF"
                  disabled={false}
                  id={null}
                />
              </Label__Label>
              <FlowItem__Content>
                <div
                  className="FlowItem__Content-bZFRMT cwqVhq"
                />
              </FlowItem__Content>
            </div>
          </FlowItem__Container>
        </FlowItem>
        <FlowItem
          alignment="stretch"
          className={null}
          error={false}
          flexibleContent={false}
          helpText={null}
          id={null}
          label={null}
          moreContent={null}
        >
          <FlowItem__Container
            alignment="stretch"
            className={null}
            id={null}
          >
            <div
              className="FlowItem__Container-krUYur jVAKAe"
              id={null}
            >
              <Label__Label
                className={null}
                disabled={false}
                id={null}
              >
                <label
                  className="Label__Label-jsURsk gdcnMF"
                  disabled={false}
                  id={null}
                />
              </Label__Label>
              <FlowItem__Content>
                <div
                  className="FlowItem__Content-bZFRMT cwqVhq"
                />
              </FlowItem__Content>
            </div>
          </FlowItem__Container>
        </FlowItem>
      </div>
    </FlowRow__BaseStyles>
  </FlowRow__Styles>
</FlowRow>
`;
